dds/idl/marshal_generator.cpp

```diff
@@ -909,6 +909,366 @@ namespace {
     }
   }
 
+#if OPENDDS_HAS_MAP
+  void gen_map_i(
+    UTL_ScopedName* tdname, AST_Map* map, bool nested_key_only, const FieldInfo* anonymous = 0)
+  {
+    // be_global->add_include("dds/DCPS/Util.h");
+    // be_global->add_include("dds/DCPS/Serializer.h");
+    if (anonymous) {
+      map = dynamic_cast<AST_Map*>(anonymous->type_);
+    }
+    const std::string named_as = anonymous ? anonymous->scoped_type_ : scoped(tdname);
+    RefWrapper base_wrapper(map, named_as, "map");
+    base_wrapper.nested_key_only_ = nested_key_only;
+
+    NamespaceGuard ng(!anonymous);
+
+    // if (!anonymous) {
+    //   for (size_t i = 0; i < array_count(special_sequences); ++i) {
+    //     if (special_sequences[i].check(base_wrapper.type_name_)) {
+    //       special_sequences[i].gen(base_wrapper.type_name_);
+    //       return;
+    //     }
+    //   }
+    // }
+
+    AST_Type* key = resolveActualType(map->key_type());
+    AST_Type* val = resolveActualType(map->value_type());
+    // TryConstructFailAction try_construct = be_global->sequence_element_try_construct(map);
```

Comment:
You will need to be able to get try_construct annotation values, I think from both the key and value types.

Reply:
What exactly are try_construct annotations?

Reply:
`@try_construct` from the XTypes spec